lib/topics_filter.rb

```diff
@@ -368,6 +370,27 @@ def tag_ids_from_tag_names(tag_names)
     all_tag_ids
   end
 
+  def filter_tag_groups(values:)
+    values.each do |key_prefix, tag_groups|
+      tag_group_ids = TagGroup.visible(@guardian).where(name: tag_groups).pluck(:id)
```

Comment:
Can you add a test case for the guardian just to make sure we don't regress in the future inadvertently?